wls-broadcast-service/src/test/java/de/muenchen/oss/wahllokalsystem/domain/MessageRepositoryTest.java

```diff
@@ -0,0 +1,105 @@
+package de.muenchen.oss.wahllokalsystem.domain;
+
+import static de.muenchen.oss.wahllokalsystem.TestConstants.SPRING_NO_SECURITY_PROFILE;
+import static de.muenchen.oss.wahllokalsystem.TestConstants.SPRING_TEST_PROFILE;
+
+import org.assertj.core.api.Assertions;
+import de.muenchen.oss.wahllokalsystem.MicroServiceApplication;
+import de.muenchen.oss.wahllokalsystem.rest.BroadcastMessageDTO;
+import de.muenchen.oss.wahllokalsystem.service.BroadcastService;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+@SpringBootTest(
+        classes = { MicroServiceApplication.class },
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+        properties = {
+                "spring.datasource.url=jdbc:h2:mem:wahllokalsystem;DB_CLOSE_ON_EXIT=FALSE",
+                "refarch.gracefulshutdown.pre-wait-seconds=0"
+        }
+)
+@ActiveProfiles(profiles = { SPRING_TEST_PROFILE, SPRING_NO_SECURITY_PROFILE })
+@Slf4j
+class MessageRepositoryTest {
+
+    @Autowired
+    BroadcastService broadcast_S;
+
+    @Autowired
+    private MessageRepository repository;
+
+    @Test
+    @Transactional(propagation = Propagation.REQUIRED, noRollbackFor = Exception.class)
+    void testSave() {
+
+        String originalOid = "1-2-3-4-5";
+        String originalNachricht = "Test Nachricht";
+
+        // Implement your logic here by replacing and/or extending the code
+
+        // initialize
+        Message original = new Message();
+        original.setOid(UUID.fromString(originalOid));
+        original.setNachricht(originalNachricht);
+
+        // persist
+        original = repository.save(original);
+
+        // check
+        Message persisted = repository.findById(original.getOid()).orElse(null);
+
+        Assertions.assertThat(persisted).isNotNull();
+        Assertions.assertThat(persisted).isEqualTo(original);
+
+    }
+
+    /**
+     * Tests if searched saved Message from many is the first found
+     */
+    @Test
+    @Transactional(propagation = Propagation.REQUIRED, noRollbackFor = Exception.class)
+    void testFirst() {
```

Comment:
Der Testcase sollte nach der zu testenden Methode im Repo benannt sein

```suggestion
    void findFirstByWahlbezirkIDOrderByEmpfangsZeit_FirstMessageIsFound() {
```

Reply:
resolved in 49e1cdd77b708dc1c8dce98e86ace0f460b9d7fe